mlflow/tracking/client.py

```diff
@@ -910,6 +914,50 @@ def log_artifacts(self, run_id, local_dir, artifact_path=None):
         """
         self._tracking_client.log_artifacts(run_id, local_dir, artifact_path)
 
+    @contextlib.contextmanager
+    def _log_artifact_helper(self, run_id, artifact_file):
+        """
+        Yields a temporary path to store a file, and then calls `log_artifact` against that path.
+        """
+        norm_path = posixpath.normpath(artifact_file)
+        filename = posixpath.basename(norm_path)
+        artifact_path = posixpath.dirname(norm_path)
+        artifact_path = None if artifact_path == "" else artifact_path
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
```

Comment:
Seems like we should clean up this directory by calling `cleanup` after logging the artifact? https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

Reply:
Found this comment stackoverflow:

https://stackoverflow.com/questions/13379742/right-way-to-clean-up-a-temporary-folder-in-python-class#comment114370145_19657028

Reply:
Oops, I missed

> On completion of the context or destruction of the temporary directory object the newly created temporary directory and all its contents are removed from the filesystem.

from the Python API doc...